scylla-server-rust/tests/data_type_service_test.rs

```diff
@@ -0,0 +1,72 @@
+#[path = "test_utils.rs"]
+mod test_utils;
+
+use prisma_client_rust::QueryError;
+use scylla_server_rust::{
+    controller::data_type_controller::DataTypeSend,
+    prisma,
+    services::{data_type_service, node_service},
+};
+use test_utils::cleanup_and_prepare;
+
+const DATA_TYPE_NAME: &str = "test";
+
+#[tokio::test]
+async fn test_get_all_datatypes() -> Result<(), QueryError> {
+    let db = cleanup_and_prepare().await?;
+
+    // ensure datatypes is empty
+    assert!(data_type_service::get_all_data_types(&db).await?.is_empty());
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_datatype_fail_upsert_no_node() -> Result<(), QueryError> {
+    let db = cleanup_and_prepare().await?;
+
+    // should fail since no node exists
+    data_type_service::upsert_data_type(
+        &db,
+        DATA_TYPE_NAME.to_owned(),
+        "hello wurld".to_owned(),
+        DATA_TYPE_NAME.to_owned(),
+    )
+    .await
+    .expect_err("Should error, as no node exists");
```

Comment:
we should be expecting the proper error that its gonna have

Reply:
Unfortunately the prisma client rust query errors are not typed.  It basically is just a wrapper for the error the database returned as a string.  Kinda sucks I know.